x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/util/CollectionUtils.java

```diff
@@ -30,12 +30,8 @@ public static <T> List<T> combine(List<? extends T> left, List<? extends T> righ
         }
 
         List<T> list = new ArrayList<>(left.size() + right.size());
-        if (left.isEmpty() == false) {
-            list.addAll(left);
-        }
-        if (right.isEmpty() == false) {
-            list.addAll(right);
-        }
```

Comment:
Both `isEmpty()` conditionas are always true at this point

Reply:
:+1: